hudi-common/src/main/java/org/apache/hudi/common/util/CommitUtils.java

```diff
@@ -134,4 +136,24 @@ private static HoodieCommitMetadata buildMetadataFromStats(List<HoodieWriteStat>
     }
     return fileIdToPath;
   }
+
+  /**
+   * Get the Metadata from the latest commit file.
+   * @param metaClient The {@link HoodieTableMetaClient} to get access to the meta data.
+   * @return An Optional {@link HoodieCommitMetadata} containing the meta data from the latest commit file.
+   */
+  public static Option<HoodieCommitMetadata> getCommitMetadataForLatestInstant(HoodieTableMetaClient metaClient) {
+    HoodieTimeline timeline = metaClient.getActiveTimeline().getCommitsTimeline().filterCompletedInstants();
```

Comment:
what if there are other completed instants on the timeline, like a compaction or clustering. they ll not have the commit offsets.

Reply:
I added code to filter and keep this to `COMMIT` for COW and `DELTACOMMIT` for MOR

Reply:
This is fixed as well